implementations/rust/ockam/ockam_core/src/routing/route.rs

```diff
@@ -13,6 +13,30 @@ pub struct Route {
     inner: VecDeque<Address>,
 }
 
+impl<A: Into<Address>, B: Into<Address>> From<(A, B)> for Route {
```

Comment:
It's ugly, but pretty common in such cases

Reply:
Hmm, so as long as there are up to 5 hops in a route .. it can be a tuple? 
It looks really nice ... but feels too hacky

Reply:
I think we can already create routes via `vec![...]`, because `Vec<Address>` implements `Into<Route>`, right? Otherwise we could have a `route!` macro which handles this conversion too (or then gives users the option to have something like this:

```rust
route![(TCP, "127.0.0.1:4000"), "my-friend"]
```

So tuples for (type, payload), or just strings if they're local addresses.

Reply:
@spacekookie creation using vec is still there, but it's sometimes not convenient. Macro would be interesting though

Reply:
I've also started thinking a `route!` macro would probably be the cleanest for end users

Reply:
So what do we think about merging this feature? The explicit unrolling of each potential size is a bit hacky, but a pretty common occurrence in rust. It does improve the end user experience in my opinion.

Reply:
@jared-s hold on, I'll try to work on macro soon